exporters/failure/collector_pagerduty.py

```diff
@@ -0,0 +1,162 @@
+            "Accept": "application/vnd.pagerduty+json;version=2",
+        }
+
+        resp = self.session.get(url, headers=headers)
+        try:
+            resp.raise_for_status()
+            # TODO too much noise?
+            logging.debug("PagerDuty successfully returned %s", resp.text)
+            return resp.json()["incidents"]
+        except requests.HTTPError as error:
+            if resp.status_code == requests.codes.unauthorized:
+                raise FailureProviderAuthenticationError from error
+            raise
+
+    def filter_by_urgency(self, urgency: str) -> bool:
+        if not self.incident_urgency:
+            return True
+        return urgency in self.incident_urgency
+
+    def filter_by_priority(self, priority: str) -> bool:
+        if not self.incident_priority:
+            return True
+        try:
+            return priority["summary"] in self.incident_priority
+        except TypeError:
+            return "null" in self.incident_priority
+
+    def search_issues(self) -> list[TrackerIssue]:
+        """
+        To maintain consistency, we are call this function `search_issues`,
+        but an `issue` in PagerDuty is called `incident`.
+        """
+        production_incidents = []
+        for incident in self.get_incidents():
+            is_production_bug = self.filter_by_urgency(
+                incident["urgency"]
+            ) and self.filter_by_priority(incident["priority"])
+
+            if is_production_bug:
+                created_at = incident["created_at"]
+                resolved_at = incident["last_status_change_at"]
+                incident_id = incident["incident_number"]
+                title = incident["title"]
+
+                created_tz = parse_assuming_utc(created_at, _DATETIME_FORMAT)
+                created_ts = second_precision(created_tz).timestamp()
+
+                resolution_tz = parse_assuming_utc(resolved_at, _DATETIME_FORMAT)
+                resolution_ts = second_precision(resolution_tz).timestamp()
+
+                if resolution_ts > created_ts:
+                    logging.debug(
+                        "Found production incident closed: {}, {}: {}".format(
+                            resolved_at,
+                            incident_id,
+                            title,
+                        )
+                    )
+                else:
+                    logging.debug(
+                        "Found production incident opened: {}, {}: {}".format(
+                            created_at,
+                            incident_id,
+                            title,
+                        )
+                    )
+                    resolution_ts = None
+
+                tracker_issue = TrackerIssue(
+                    str(incident_id),
+                    created_ts,
+                    resolution_ts,
+                    incident["service"]["summary"],
+                    # TODO another thing I thought: we could filter services
```

Comment:
Lets talk about this before merging

Reply:
Yeah it may be a valid request, but we don't know, so worth notice imo here.